PADV-1557 Adjust storage backend definition via site configurations (#9)
Pearson-Advance/edx-sga: docs/001-custom-features.md

- Find the assignment and grade it.
- After grading the assignment, it should not be necessary to approve the grade.
- Go to the progress page, with the student user and check if the grade is assigned to the unit containing the SGA component.

## Custom Storage Backend

### Context

The SGA Xblock in its upstream repository approaches the storage capability by relaying on the Django's default storage

> that is defined in the edx platform. Tehre's also a feature where the storage backend to be used can be defined via

Reviewer: ```suggestion
that is defined in the edx platform. There's also a feature where the storage backend to be used can be defined via
```

platform settings. For our fork, this feature adds the capability to define a site-aware backend via site configurations.

### Feature

> To define a d backend shall follow the following format by defining the storage class and its kwargs.

Reviewer: ```suggestion
To define a backend shall follow the following format by defining the storage class and its kwargs.
```

> To define a desire backend, follow the following format by defining the storage class and its kwargs.

Reviewer: ```suggestion
To define a desired backend, follow the following format by defining the storage class and its kwargs.
```

Author: This is weird, I know you've pointed this two times and I was pretty sure I'd push it to the PR. But seems it didn't happen. Comment applied.

> To define a desire backend shall follow the following format by defining the storage class and it's kwargs.

Reviewer: ```suggestion
To define a desired backend shall follow the following format by defining the storage class and its kwargs.
```

As an example, below you will be find de Site Configuration definition to use an AWS S3 bucket:

"SGA_STORAGE_SETTINGS": {
    "STORAGE_CLASS": "storages.backends.s3boto3.S3Boto3Storage",
    "STORAGE_KWARGS": {
        "access_key": "aws_access_key",
        "secret_key": "aws_secret_key",
        "bucket_name": "aws-bucket-name",
        "region_name": "us-east-1"
    }
}

Once it has been done, the SGA xblocks for the given site will manage the media objects with the defined storage backend.

> If no settings are defined, the Xblock will use the default Django storage

Reviewer: ```suggestion
If no settings are defined, the Xblock will use the default Django storage.
```

> If no settings are defined, the Xblock would use the default Django storage

Reviewer: ```suggestion
If no settings are defined, the Xblock will use the default Django storage.
```
